paymentsheet/src/main/java/com/stripe/android/paymentsheet/verticalmode/ManageScreenInteractor.kt

```diff
@@ -0,0 +1,28 @@
+package com.stripe.android.paymentsheet.verticalmode
+
+import com.stripe.android.model.PaymentMethod
+import com.stripe.android.model.PaymentMethodCode
+import com.stripe.android.paymentsheet.viewmodels.BaseSheetViewModel
+import com.stripe.android.ui.core.cbc.CardBrandChoiceEligibility
+import com.stripe.android.uicore.utils.mapAsStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+internal interface ManageScreenInteractor {
+    val state: StateFlow<State>
+
+    data class State(
+        val paymentMethods: List<PaymentMethod>?,
+        val nameProvider: (PaymentMethodCode?) -> String,
+        val isCbcEligible: Boolean,
+    )
+}
+
+internal class DefaultManageScreenInteractor(viewModel: BaseSheetViewModel) : ManageScreenInteractor {
+    override val state = viewModel.paymentMethods.mapAsStateFlow { paymentMethods ->
+        ManageScreenInteractor.State(
+            paymentMethods,
+            viewModel::providePaymentMethodName,
+            viewModel.paymentMethodMetadata.value?.cbcEligibility is CardBrandChoiceEligibility.Eligible,
```

Comment:
In practice, this doesn't matter, but in theory we should make this a combineAsStateFlow, and combine both the paymentMethods and paymentMethodMetadata. We don't want to be reaching into the view model for this, since it could change (even though in practice it won't).